### PR TITLE
docs: add release documentation

### PR DESCRIPTION
**Pull request by carlosmn**
This should provide the release manager enough to know which steps to take when
it's time to cut a new release.

### CODE REVIEW

File: docs/relase.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	- [ ] Bump the versions in the headers
+    - [ ] Make a release candidate
+    - [ ] Plug any final leaks
+    - [ ] Fix any last-minute issues
+    - [ ] Make sure CHANGELOG reflects everything worth discussing
+    - [ ] Update the version in CHANGELOG and the header
+    - [ ] Produce a release candidate
+    - [ ] Tag
+    - [ ] Create maint/v0.X
+    - [ ] Update any bindings the core team works with
```
Comment (pks-t) [inline docs/relase.md]: Are these additional spaces for each except the first item intentionally?

File: docs/relase.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+We tag at least one release candidate. This RC must carry the new version in the headers. If there are no significant issues found, we can go straight to the release after a single RC. This is up to the discretion of the release manager.
```
Comment (pks-t) [inline docs/relase.md]: Do we have a minimum cooking time for the RC?

File: docs/relase.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+followed by the three sections in the changelog. For release candidates we can avoid copying the full changelog and only include any new entries.
+
+During the freeze, and certainly after the first release candidate, any bindings the core team work with should be updated in order to discover any issues that might come up with the multitude of approaches to memory management, embedding or linking.
+
+Create a branch `maint/v0.X` at the current state of `master` after you've created the tag. This will be used for maintenance releases and lets our dependents track the latest state of the series.
+
+## Maintenance release
+
+Every once in a while, when we feel we've accumulated backportable fixes in the mainline branch, we produce a maintenance release in order to provide updates for those who track the releases and so our users and integrators don't have to upgrade immediately.
```
Comment (pks-t) [inline docs/relase.md]: "don't have to upgrade immediately". You probably refer to having to upgrade to the new full version? Should maybe rephrase that a bit to make that intention clear.

File: docs/relase.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+As a rule of thumb, it's a good idea to produce a maintenance release when we're getting ready for a full release. This gives a last round of fixes without having to upgrade (which with us potentially means adjusting to API changes).
```
Comment (pks-t) [inline docs/relase.md]: Do we have a policy on which versions get backports?

File: docs/relase.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    Release v0.X.Y
+    
+    Enough fixes have accumulated, let's release v0.X.Y
+    
+    - [ ] Select the changes we want to backport
+    - [ ] Update maint/v0.X
+    - [ ] Tag
+
+The list of changes to backport does not need to be comprehensive and we might not backport something if the code in mainline has diverged significantly.
```
Comment (pks-t) [inline docs/relase.md]: We should explicitly state that we will never include ABI- or API-breaking changes.

**Comment by pks-t** [comment]
One more thing just came to my mind: this does not yet state how to update our Doxygen documentation.

File: docs/relase.md (hunk 1 above)
Comment (carlosmn) [reply docs/relase.md]: I think I do it instinctively to make a list even though that's not actually a thing.

Comment (carlosmn) [reply docs/relase.md]: Ok no I see what you're saying. I'm using four spaces here so it's treated as a markdown preformatted block, as I want to show the raw markdown in the rendered text. The tab must have snuck in in some auto-reindentation.

**Comment by carlosmn** [comment]
I think I've tackled all your comments. I'm not too happy with the wording around maintenance releases as the first two paragraphs are basically saying the same.

File: docs/relase.md [hunk 6]
```diff
@@ -60,3 +60,15 @@ Here we do not use release candidates as the changes are supposed to be small an
...
+
+We use docurium to generate our documentation. It is a tool written in ruby which leverages libclang's documentation parser. Install docurium
+
+    gem install docurium
+
+and run it against our description file with the tip of master checked out.
+
+    cm doc api.docurium
+
+It will start up a few proceses and write out the results as a new commit onto the `gh-pages` branch. That can be pushed to GitHub to update what will show up on our documentation reference site.
```
Comment (pks-t) [inline docs/relase.md]: Typo: "proceses".

I guess for this step there's no need for a separate PR?

File: docs/relase.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
+# Releasing the library
```
Comment (tiennou) [inline docs/relase.md]: typo: `docs/relase.md`

Comment (pks-t) [reply docs/relase.md]: Will fix in a separate PR, thanks!
